### ISSUE TITLE
dotnet publish no longer forwards the path the project it being published to (regression)

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#1014]
The publish command used to pass the path the project was published to in the %publish:OutputPath% parameter. After this commit https://github.com/dotnet/cli/commit/ac2d0e36a9b1353f3461c01f60567d9dc541e0eb#diff-06bec5144cd01c123b4fd6d6c7f4c027R92 it no longer does. As a result pre/post publish scripts are useless since it they don't know where the project was published.

Note it is impossible to figure out the path where the project is - `%publish.OutputPath%/%publish:Configuration%/%publish:Framework%` results in `C:\Temp\WebApp6IIS/Debug/DNXCore` but the project was actually published to `C:\Temp\WebApp6IIS\Debug\dnxcore50`

**Comment by piotrpMSFT** [comment]
Project publish produces multiple paths. Currently it is count(TFM). How do you propose passing these?

**Comment by piotrpMSFT** [comment]
@livarcocc

**Comment by moozzyk** [comment]
@piotrpMSFT - I spent some time with dotnet publish in the last few days and I actually liked the way it worked before much more (except that it would by default publish for all frameworks and the artifacts would stomp on each other). Why not make providing the framework mandatory (at least when output path is provided) and if the user wants to publish for more than one framework they would call it twice?
I am not sure which flow is more common publishing for all frameworks the project supports or publishing for just one...

**Comment by livarcocc** [comment]
@moozzyk First of all, I have opened #1016  to track why %publish:Framework% is not being consistent and giving the right framework name. Feels to me like a separate bug. Second, I believe the fix that you submitted here https://github.com/dotnet/cli/pull/1015 makes more sense.

And third, while I believe your suggestion is a great one, I believe requiring framework with the output path adds to the complexity of the command and makes for a less than ideal experience. We did discuss the possibility of when passing a framework, to not append the config/tfm to the output path, but in the end we decided for consistency with the other commands and again thought that having less of these rules in the command would make for a cleaner experience.
